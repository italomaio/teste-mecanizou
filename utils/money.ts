export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
