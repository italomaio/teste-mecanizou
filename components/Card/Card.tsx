import React from "react";

import { cn } from "@/utils/classes";

const Card = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(
      "bg-white shadow-md rounded-md p-4 dark:bg-gray-800 dark:text-white",
      className
    )}
    {...props}
  />
));

Card.displayName = "Card";

export default Card;
