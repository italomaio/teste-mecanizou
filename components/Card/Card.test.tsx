import React from "react";
import Card from "./Card";
import { render } from "@/tests/utils";

describe("Card Component", () => {
  it("Should render without crashing", () => {
    const { container } = render(<Card />);
    expect(container.firstChild).toBeInTheDocument();
  });

  it("Should apply custom className", () => {
    const { container } = render(<Card className="custom-class" />);
    expect(container.firstChild).toHaveClass("custom-class");
  });

  it("Should forward ref to the underlying div", () => {
    const ref = React.createRef<HTMLDivElement>();
    render(<Card ref={ref} />);
    expect(ref.current).not.toBeNull();
  });

  it("Should render children correctly", () => {
    const { getByText } = render(<Card>Test Content</Card>);
    expect(getByText("Test Content")).toBeInTheDocument();
  });

  it("Should apply default styles", () => {
    const { container } = render(<Card />);
    expect(container.firstChild).toHaveClass(
      "bg-white shadow-md rounded-md p-4 dark:bg-zinc-800 dark:text-white dark:border dark:border-zinc-700 dark:shadow-none"
    );
  });

  it("Should match the snapshot", () => {
    const { asFragment } = render(<Card>Snapshot Content</Card>);
    expect(asFragment()).toMatchSnapshot();
  });
});
