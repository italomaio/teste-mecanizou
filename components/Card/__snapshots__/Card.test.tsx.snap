// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Card Component Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white shadow-md rounded-md p-4 dark:bg-zinc-800 dark:text-white dark:border dark:border-zinc-700 dark:shadow-none"
  >
    Snapshot Content
  </div>
</DocumentFragment>
`;
